<launch>
    <!-- Set use_sim_time true to generate trajectory bag file in same time domain -->
    <param name="/use_sim_time" value="true"/>

    <!-- Path of a rosbag file -->
    <arg name="bag_path" default="$(find path_recorder)/../bag/nclt_2011_09_30_drive_0027_synced"/>
   <!-- For Velodyne VLP-16 -->
   <!-- <param name="scan_line" value="16" /> -->
    

   <!-- For Velodyne HDL-32 -->
   <!-- <param name="scan_line" value="32" /> -->
   

   <!-- For Velodyne HDL-64 -->
   <param name="scan_line" value="64" />
   

    <!-- Run Faster_lio -->
    <group>
        <arg name="rviz" default="true" />

        <rosparam command="load" file="$(find faster_lio)/config/velodyne.yaml" />
    
        <param name="feature_extract_enable" type="bool" value="0"/>
        <param name="point_filter_num_" type="int" value="4"/>
        <param name="max_iteration" type="int" value="3" />
        <param name="filter_size_surf" type="double" value="0.5" />
        <param name="filter_size_map" type="double" value="0.5" />
        <param name="cube_side_length" type="double" value="1000" />
        <param name="runtime_pos_log_enable" type="bool" value="1" />
        <node pkg="faster_lio" type="run_mapping_online" name="laserMapping" output="screen" /> 
    
        <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find faster_lio)/rviz_cfg/loam_livox.rviz" />
        </group>
    </group>


	<!-- Remap point cloud topic -->
    <node pkg="kiss_icp_kitti" type="kitti_helper_node" name="kitti_helper_node" output="screen"/>
	<!-- Play nclt rosbag file -->
    <node pkg="rosbag" type="play" name="nclt" args="--clock $(arg bag_path)/kitti.bag " output="log" />

	<!-- Launch path_recorder to save a trajectory from KISS-ICP -->
    <!-- <node pkg="path_recorder" type="path_recorder" name="kiss_recorder" output="screen">
        <param name="algorithm_name" type="string" value="faster_lio" />
        <param name="path_topic" type="string" value="/path" />
        <param name="save_to" type="string" value="$(arg bag_path)" />
    </node> -->
</launch>
